app/models/calagator/venue/geocoder.rb

```diff
@@ -13,26 +10,36 @@ def self.geocode(venue)
 
     def geocode
       return true unless should_geocode?
-
-      self.geo = Geokit::Geocoders::MultiGeocoder.geocode(venue.geocode_address)
-      if geo.success
-        venue.latitude       = geo.lat
-        venue.longitude      = geo.lng
-        venue.street_address = geo.street_address if venue.street_address.blank?
-        venue.locality       = geo.city           if venue.locality.blank?
-        venue.region         = geo.state          if venue.region.blank?
-        venue.postal_code    = geo.zip            if venue.postal_code.blank?
-        venue.country        = geo.country_code   if venue.country.blank?
-      end
-
+      map_geo_to_venue if geo.success
       log
     end
 
     private
 
+    def geo
+      @geo ||= Geokit::Geocoders::MultiGeocoder.geocode(venue.geocode_address)
+    end
+
+    VENUE_GEO_FIELD_MAP = {
+      latitude:       :lat,
+      longitude:      :lng,
+      street_address: :street_address,
+      locality:       :city,
+      region:         :state,
+      postal_code:    :zip,
+      country:        :country_code,
+    }
+
+    def map_geo_to_venue
+      VENUE_GEO_FIELD_MAP.each do |venue_field, geo_field|
+        next if venue[venue_field].present?
```

Comment:
This deviates from the original implementation, in that we used to write `lat` and `lng` even when values were present on the venue. I think this is required for the "force geocoding" feature to work correctly.